app/controllers/shared/api.php

```diff
@@ -14,10 +14,38 @@
 use Appwrite\Extend\Exception;
 use Utopia\Abuse\Abuse;
 use Utopia\Abuse\Adapters\TimeLimit;
+use Utopia\Cache\Adapter\Filesystem;
+use Utopia\Cache\Cache;
 use Utopia\Database\Database;
 use Utopia\Database\Document;
 use Utopia\Database\Validator\Authorization;
-use Utopia\Registry\Registry;
+
+$parseLabel = function (string $label, array $responsePayload, array $requestParams, Document $user) {
+    preg_match_all('/{(.*?)}/', $label, $matches);
+    foreach ($matches[1] ?? [] as $pos => $match) {
+        $find = $matches[0][$pos];
+        $parts = explode('.', $match);
+
+        if (count($parts) !== 2) {
+            throw new Exception('Too less or too many parts', 400, Exception::GENERAL_ARGUMENT_INVALID);
+        }
+
+        $namespace = $parts[0] ?? '';
+        $replace = $parts[1] ?? '';
+
+        $params = match ($namespace) {
+            'user' => (array)$user,
+            'request' => $requestParams,
+            default => $responsePayload,
+        };
+
+        if (array_key_exists($replace, $params)) {
+            $label = \str_replace($find, $params[$replace], $label);
+        }
+    }
+
+    return $label;
+};
```

Comment:
wait this same change is present in the abuse PR right ? how did it make it to this PR ?

Reply:
They both share this function